lib/attr_encrypted.rb

```diff
@@ -341,7 +341,8 @@ def load_iv_for_attribute(attribute, algorithm)
 
       def load_salt_for_attribute(attribute)
         encrypted_attribute_name = self.class.encrypted_attributes[attribute.to_sym][:attribute]
-        salt = send("#{encrypted_attribute_name}_salt") || send("#{encrypted_attribute_name}_salt=", SecureRandom.hex)
+        salt = send("#{encrypted_attribute_name}_salt") || send("#{encrypted_attribute_name}_salt=", [SecureRandom.random_bytes].pack("m"))
+        salt = salt.size > 16 ? salt.unpack("m").first : salt
```

Comment:
I can confirm this will work, at least for our deployment. What's not clear to me is why existing salt values are only 16 characters long, since ```SecureRandom.hex``` [returns a string of length 32](http://ruby-doc.org/stdlib-2.2.0/libdoc/securerandom/rdoc/SecureRandom.html#method-c-hex) when called without arguments.

If it's possible that different deployments will store different sized hex salt, a safer approach might be to store new salt values prefixed with an underscore (or any other non-hex character) and strip before unpacking.

If we can confirm that existing values are always 16 characters, on the other hand, then this approach is solid. Just stumped as to why they're not 32 characters (16 bytes of hex). The [default cypher](https://github.com/attr-encrypted/encryptor/blob/master/lib/encryptor.rb#L18), ```aes-256-cbc```, only uses 16 bytes of salt, but I don't see it truncating the supplied value anywhere.